tools/cmake/cbmc_recipes.cmake

```diff
@@ -0,0 +1,323 @@
+)
+
+add_custom_command(
+    COMMENT "Dropping unused functions from ${cbmc_proof_name}"
+    DEPENDS
+        ${cbmc_proof_name}_0040_library_added.goto
+    OUTPUT
+        ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+    COMMAND
+        goto-instrument --drop-unused-functions --verbosity ${CBMC_VERBOSITY}
+        ${cbmc_proof_name}_0040_library_added.goto
+        ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+)
+
+list(LENGTH cbmc_nondet_static_exclude cnse_length)
+if(${cbmc_nondet_static} OR ${cnse_length})
+    if(${cbmc_nondet_static})
+        set(nondet_static_flags --nondet-static)
+    else()
+        list(JOIN
+            cbmc_nondet_static_exclude
+            ";--nondet-static-exclude;" nondet_static_flags)
+        list(PREPEND
+            nondet_static_flags
+            --nondet-static-exclude
+        )
+    endif()
+
+    add_custom_command(
+        COMMENT
+            "Havoccing all except ${cnse_length} statics in ${cbmc_proof_name}"
+        DEPENDS
+            ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+        OUTPUT
+            ${cbmc_proof_name}_0060_nondet_static.goto
+        COMMAND
+            goto-instrument ${nondet_static_flags}
+            --verbosity ${CBMC_VERBOSITY}
+            ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+            ${cbmc_proof_name}_0060_nondet_static.goto
+    )
+else()
+    add_custom_command(
+        COMMENT
+            "Not havoccing statics in ${cbmc_proof_name}"
+        DEPENDS
+            ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+        OUTPUT
+            ${cbmc_proof_name}_0060_nondet_static.goto
+        COMMAND
+            cmake -E copy
+            ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+            ${cbmc_proof_name}_0060_nondet_static.goto
+    )
+endif()
+
+add_custom_command(
+    COMMENT "Removing unused globals initialization for ${cbmc_proof_name}"
+    DEPENDS
+        ${cbmc_proof_name}_0060_nondet_static.goto
+    OUTPUT
+        ${cbmc_proof_name}.goto
+    COMMAND
+        goto-instrument --slice-global-inits --verbosity ${CBMC_VERBOSITY}
+        ${cbmc_proof_name}_0060_nondet_static.goto
+        ${cbmc_proof_name}.goto
+)
+
+add_custom_target(${cbmc_proof_name}-proof DEPENDS ${cbmc_proof_name}.goto)
+
+
+# ______________________________________________________________________________
+#     Common recipe for running a single proof using CTest.
+# ``````````````````````````````````````````````````````````````````````````````
+
+list(APPEND cbmc_flags
+    --object-bits ${cbmc_object_bits}
+    --unwind ${cbmc_unwind}
+)
+
+list(LENGTH cbmc_unwindset cbmc_unwindset_length)
+if(${cbmc_unwindset_length} GREATER 0)
+    list(JOIN cbmc_unwindset "," _cbmc_unwindset)
+    list(APPEND cbmc_flags --unwindset ${_cbmc_unwindset})
+endif()
+
+list(JOIN cbmc_flags " " _cbmc_flags)
+
+# Actually check the proof
+# TODO this obviously won't work on windows.
+add_test(
```

Comment:
Assuming the rest of this is going to be replaced with your new implementation that is OS agnostic.

Reply:
done